portal/patient/libs/Controller/OnsiteDocumentController.php

```diff
@@ -379,4 +384,21 @@ public function Delete()
             $this->RenderExceptionJSON($ex);
         }
     }
+
+    /**
+     * @param $encoded
+     * @param $v
+     * @return bool|string
+     */
+    private function decode($encoded, $v): bool|string
+    {
+        $encoded = base64_decode($encoded);
+        $decoded = "";
+        for ($i = 0; $i < strlen($encoded); $i++) {
+            $b = ord($encoded[$i]);
+            $a = $b ^ $v;
+            $decoded .= chr($a);
+        }
+        return base64_decode(base64_decode($decoded));
```

Comment:
hi @sjpadgett , why would you `base64_decode` twice?

Reply:
Check out the bitwise.